File: LRP/LocalCSharpServer/Simple.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SoftFX.Lrp;

namespace LocalCSharp
{
	internal class Simple
	{
		internal LPtr Constructor()
		{
			return LPtr.Zero;
		}
		internal string Inverse(string text)
		{
			string result = AppDomain.CurrentDomain.ToString();
			return result;
		}
		internal bool Factorial(int value, out int result)
		{
			result = 0;
			if (value < 0)
			{
				return false;
			}
			int answer = 1;
			while (value > 1)
			{
				answer *= value;
				--value;
				if (result < 0)
				{
					return false;
				}
			}
			result = answer;
			return true;
		}
	}
}

File: LRP/LocalCppServer/Simple.cpp
#include "stdafx.h"
#include "Simple.h"

string CSimple::Inverse(const string& text)
{
	string result(text.rbegin(), text.rend());
	return result;
}

bool CSimple::Factorial(int value, int& result)
{
	if (value < 0)
	{
		return false;
	}
	int answer = 1;
	while(value > 1)
	{
		answer *= value;
		--value;
		if (result < 0)
		{
			return false;
		}
	}
	result = answer;
	return true;
}
void* CSimple::Constructor()
{
	return nullptr;
}
namespace
{
	CSimple gSimple;
}

CSimple& GetSimple()
{
	return gSimple;
}
